src/main/java/org/isf/sms/providers/skebby/SkebbyGatewayService.java

```diff
@@ -0,0 +1,150 @@
+	private static final String RESPONSE_SUCCESS = "OK";
+	private static final String KEY_PASSWORD = "skebby-gateway-service.password";
+	private static final String KEY_USERNAME = "skebby-gateway-service.username";
+	private static final String KEY_MESSAGE_TYPE = "skebby-gateway-service.message-type";
+
+	private static final String KEY_USER_KEY = "skebby-gateway-service.userKey";
+	private static final String KEY_ACCESS_TOKEN = "skebby-gateway-service.accessToken";
+
+	private static final String KEY_SENDER = "skebby-gateway-service.sender";
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(SkebbyGatewayService.class);
+
+	@Resource(name = "smsProperties")
+	private Properties smsProperties;
+
+	@Autowired
+	private SkebbyGatewayConverter skebbyGatewayConverter;
+
+	@Override
+	public boolean sendSMS(Sms sms) {
+		String userKeyAccessToken = loginUserKeySessionKey();
+		String userKey = SkebbyGatewayUtil.extractUserKey(userKeyAccessToken);
+		String sessionKeyOrAccessToken = SkebbyGatewayUtil.extractSessionKey(userKeyAccessToken);
+		MessageType messageType = null;
+
+		final String msgType = this.smsProperties.getProperty(KEY_MESSAGE_TYPE);
+		if (msgType != null) {
+			messageType = MessageType.valueOf(msgType);
+		}
+
+		final String sender = this.smsProperties.getProperty(KEY_SENDER);
+
+		SckebbySmsRequest smsSendingRequest = this.skebbyGatewayConverter.toServiceDTO(sms, messageType, sender);
+
+		SkebbyGatewayRemoteService httpClient = buildHttlClient();
+		System.out.println("Sending...");
+		SckebbySmsResponse result = null;
+		try {
+			if (this.isAccessTokenAuthentication()) {
+				result = httpClient.sendSmsWithAccessToken(userKey, sessionKeyOrAccessToken, smsSendingRequest).getBody();
+			} else {
+				result = httpClient.sendSmsWithSessionKey(userKey, sessionKeyOrAccessToken, smsSendingRequest).getBody();
+			}
+		} catch (FeignException fe) {
+			// skebby replies with HTTP 400 when you have not more sms, so that we could have an exception
+			LOGGER.error("Gateway error!");
+			LOGGER.error(fe.getMessage());
+			return false;
+		}
+		return result != null && RESPONSE_SUCCESS.equals(result.getResult());
+	}
+
+	private SkebbyGatewayRemoteService buildHttlClient() {
+		String baseUrl = this.smsProperties.getProperty(this.getRootKey() + ".ribbon.base-url");
+		return Feign.builder().encoder(new CustomCommonEncoder()).decoder(new CustomCommonDecoder()).logger(new Slf4jLogger(SkebbyGatewayRemoteService.class))
+						.logLevel(feign.Logger.Level.FULL).contract(new SpringMvcContract()).target(SkebbyGatewayRemoteService.class, baseUrl);
+	}
+
+	private String loginUserKeySessionKey() {
+		// USER_KEY and ACCESS_TOKEN avoids the login call every time we need to send sms
+		final String userKey = this.smsProperties.getProperty(KEY_USER_KEY);
+		final String token = this.smsProperties.getProperty(KEY_ACCESS_TOKEN);
+		if (userKey != null && !userKey.isBlank() && token != null && !token.isBlank()) {
```

Comment:
isBlank has been introduced with JDK11 so the maven build fails for JDK8.
Maybe we can replace with .trim().isEmpty()?

Reply:
Done locally, will be pushed with the merge commit